action.yml

```diff
@@ -23,23 +24,72 @@ runs:
+        # Try to get the latest release from GitHub API with proper error handling
+        if ! TERMINUS_RELEASE=$(curl --silent --fail --max-time 10 \
             --header 'authorization: Bearer ${{ github.token }}' \
             "https://api.github.com/repos/pantheon-systems/terminus/releases/latest" \
-            | perl -nle'print $& while m#"tag_name": "\K[^"]*#g'
-        )
+            | perl -nle'print $& while m#"tag_name": "\K[^"]*#g'); then
+          echo "::error::Failed to connect to GitHub API to determine latest Terminus version."
+          echo "::error::Please specify a terminus-version input parameter for deterministic builds."
+          exit 1
+        fi
+
+        # Validate we got a proper version string (should be semver format)
+        if [[ -z "$TERMINUS_RELEASE" || ! "$TERMINUS_RELEASE" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+          echo "::error::Unable to determine a valid Terminus version from GitHub API."
+          echo "::error::Please specify a terminus-version input parameter for deterministic builds."
+          exit 1
+        fi
+
+        echo "Latest Terminus version detected: $TERMINUS_RELEASE"
         echo "TERMINUS_RELEASE=$TERMINUS_RELEASE" >> $GITHUB_ENV
 
     - name: Install Terminus
       shell: bash
       run: |
-        mkdir $HOME/terminus && cd $HOME/terminus
+        mkdir -p $HOME/terminus && cd $HOME/terminus
         echo "Installing Terminus v$TERMINUS_RELEASE"
-        curl -L https://github.com/pantheon-systems/terminus/releases/download/$TERMINUS_RELEASE/terminus.phar --output terminus
+
+        # Download Terminus with validation
+        if ! curl --fail -L https://github.com/pantheon-systems/terminus/releases/download/$TERMINUS_RELEASE/terminus.phar --output terminus; then
+          echo "::error::Failed to download Terminus v$TERMINUS_RELEASE"
+          exit 1
+        fi
+
+        # Download checksum for verification
+        if curl --fail -L https://github.com/pantheon-systems/terminus/releases/download/$TERMINUS_RELEASE/terminus.phar.sha256 --output terminus.sha256; then
+          echo "Verifying download integrity..."
+          # Fix the checksum file format if needed (ensure it only contains hash and filename)
+          sed -i.bak 's/^.*\([0-9a-f]\{64\}\).*$/\1  terminus/g' terminus.sha256
+          if ! sha256sum -c terminus.sha256; then
+            echo "::error::Checksum verification failed! The downloaded file may be corrupted."
+            exit 1
+          fi
+          echo "Integrity verification passed."
+        else
+          echo "::warning::Could not download checksum file. Skipping integrity check."
+        fi
+
+        # Basic file type verification
+        file_info=$(file terminus)
+        if ! echo "$file_info" | grep -q -E 'PHP script|executable|.+ASCII.+PHP'; then
+          echo "::error::Downloaded file is not a PHP script or executable. Got: $file_info"
+          exit 1
+        fi
```

Comment:
File type check should be removed. The output of the file command is not consistent across systems, and it's clear you've already encountered this since the grep is attempting to match 3 different possible responses.

It is receiving a different response in Github Actions than what the pattern match currently covers. Adding more conditions to account for this additional format will still not provide full coverage. Additionally it assumes that the file command exists, which will not always be the case.

This check is also made redundant by the terminus --version test at the end. That test will not succeed if what's downloaded isn't the correct type.